src/api/records.tsx

```diff
@@ -585,15 +625,22 @@ export const useIncomingRecordCount = (
     number,
     AxiosError,
     number,
-    [string, { searchParams: SearchParams; filters: string[] }]
+    [
+      string,
+      { searchParams: SearchParams; filters: string[]; projection: string[] },
+    ]
   >(
     [
       'incomingRecordCount',
-      { searchParams: finalisedSearchParams, filters: finalisedFilters },
+      {
+        searchParams: finalisedSearchParams,
+        filters: finalisedFilters,
+        projection,
```

Comment:
I think we need to remove the projection from the incoming record count query, as otherwise we'd need to check the user isn't querying for too many rows whenever they change the data channels they're viewing.

A slight optimisation we can do is to keep the `projection` key here in the query key, but set it to `projection: [timeChannelName]` - this way we  can still optimise the `recordCount` query in the "blank table" use case, but otherwise we don't use the projection elsewhere in `incomingRecordCount`